src/Gui/NavigationAnimation.cpp

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: LGPL-2.1-or-later
+/****************************************************************************
+ *                                                                          *
+ *   Copyright (c) 2023 Bas Ruigrok (Rexbas) <Rexbas@proton.me>             *
+ *                                                                          *
+ *   This file is part of FreeCAD.                                          *
+ *                                                                          *
+ *   FreeCAD is free software: you can redistribute it and/or modify it     *
+ *   under the terms of the GNU Lesser General Public License as            *
+ *   published by the Free Software Foundation, either version 2.1 of the   *
+ *   License, or (at your option) any later version.                        *
+ *                                                                          *
+ *   FreeCAD is distributed in the hope that it will be useful, but         *
+ *   WITHOUT ANY WARRANTY; without even the implied warranty of             *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU       *
+ *   Lesser General Public License for more details.                        *
+ *                                                                          *
+ *   You should have received a copy of the GNU Lesser General Public       *
+ *   License along with FreeCAD. If not, see                                *
+ *   <https://www.gnu.org/licenses/>.                                       *
+ *                                                                          *
+ ***************************************************************************/
+
+#include "PreCompiled.h"
+#include "NavigationAnimation.h"
+#include <Inventor/nodes/SoCamera.h>
+
+using namespace Gui;
+
+NavigationAnimation::NavigationAnimation(NavigationStyle* navigation)
+    : navigation(navigation)
+{}
+
+void NavigationAnimation::updateCurrentValue(const QVariant& value)
+{
+    if (state() == QAbstractAnimation::State::Stopped) {
+        return;
+    }
+    update(value);
+}
+
+void NavigationAnimation::onStop(bool finished)
```

Comment:
Unused parameter: Use `Q_UNUSED` or `[[maybe_unused]]`

Reply:
Resolved in 2fec778d31a44790854e6ea224d0a436664d8096